core/src/main/java/org/elasticsearch/index/analysis/CustomAnalyzerProvider.java

```diff
@@ -88,6 +94,34 @@ public void build(final Map<String, TokenizerFactory> tokenizers, final Map<Stri
         );
     }
 
+    public static TokenFilterFactory checkAndApplySynonymFilter(TokenFilterFactory tokenFilter, String tokenizerName, TokenizerFactory tokenizer,
+                                                                List<TokenFilterFactory> tokenFilterList,
+                                                                List<CharFilterFactory> charFiltersList,
+                                                                int positionIncrementGap, int offsetGap, Environment env) {
```

Comment:
this method does not need the gaps I think? Let's remove them from the signature and use the defaults in the body of this method?